### PR TITLE
add xps 17 9710

### PR DESCRIPTION
**Pull request by NelsonJeppesen**
# Add XPS 17 9710

Mostly just to share the fix for non-working sound

Tested with Kernel 5.15.12 on the non-nvidia xps17

### CODE REVIEW

File: dell/xps/17-9710/intel/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    ../../../../common/pc/ssd
+  ];
+
+  # patch the kernel to enable sound
+  # Tested with 5.15.12
+  boot.kernelPatches = lib.mkDefault [{
+    name = "xps17-9710-fix-sound";
+    patch = null;
+    extraConfig = ''
+      SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES y
```
Comment (Mic92) [inline dell/xps/17-9710/intel/default.nix:14]: How does this option fix sound? https://cateee.net/lkddb/web-lkddb/SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES.html

Comment (Mic92) [reply dell/xps/17-9710/intel/default.nix:14]: Ok. get it. It's a dependency for `SND_SOC_INTEL_SOUNDWIRE_SOF_MACH`.
Can you make a PR for nixpkgs to include both options in nixpkgs master?

File: dell/xps/17-9710/intel/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }];
+
+  # Includes the Wi-Fi and Bluetooth firmware
+  hardware.enableRedistributableFirmware = true;
+
+  # Requires at least 5.12 for working sound
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.12") (lib.mkDefault pkgs.linuxPackages_latest);
+
+  # Allows for updating firmware via `fwupdmgr`.
+  services.fwupd.enable = true;
```
Comment (Mic92) [inline dell/xps/17-9710/intel/default.nix:26]: Maybe fwupd is supported by this device in the README instead and leave it to the user if they want to enable it permanently or not. Without documentation that this service exists, it wouldn't be very useful anyway.

Comment (NelsonJeppesen) [reply dell/xps/17-9710/intel/default.nix:26]: :+1: Sounds reasonable; I'll take this as an excuse to create readme

File: dell/xps/17-9710/intel/default.nix (hunk 1 above)
Comment (NelsonJeppesen) [reply dell/xps/17-9710/intel/default.nix:14]: Add it to `nixpkgs`, like here? https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/common-config.nix#L290

Comment (Mic92) [reply dell/xps/17-9710/intel/default.nix:14]: Yes.

Comment (NelsonJeppesen) [reply dell/xps/17-9710/intel/default.nix:14]: @Mic92 I've created and tested a nixpkgs pr locally
https://github.com/NixOS/nixpkgs/pull/154220

**Comment by NelsonJeppesen** [comment]
Moving to draft, pending approval or rejection of nixpkgs pr
https://github.com/NixOS/nixpkgs/pull/154220

**Comment by Mic92** [comment]
Now we just need to drop the kernel options from this PR: https://github.com/NixOS/nixpkgs/pull/154287

**Comment by NelsonJeppesen** [comment]
Added simple wiki
Removed kernel patch

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**